### DISCUSSION TITLE
Override macOS menu keybinds

### DISCUSSION THREAD

**Discussion by bentinata** [wezterm/wezterm#6427] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

I would like to rebind `Cmd+K` to `ClearScrollbackAndViewport` but unable to do so.

Without any keybinds (`weztern -n`), my macOS Edit -> Clear the scrollback and viewport is blank. Edit menu looks like this:

<img width="400" alt="Screenshot 2024-11-22 at 11 55 14" src="https://github.com/user-attachments/assets/b9214abf-e694-442a-8f63-6779b3fbf2d2">

I assume this is because by default it is assigned to nothing.
https://github.com/wez/wezterm/blob/4050072da21cc3106d0985281d75978c07e22abc/wezterm-gui/src/commands.rs#L774-L781

If I define my `wezterm.lua` like this:

```lua
return {
	// other configs ...
	keys = {
		{
			key = "K",
			mods = "CMD",
			action = wezterm.action.ClearScrollback("ScrollbackAndViewport"),
		},
	},
}
```

It will not replace the `Cmd+K` assigned Edit > Clear scrollback and fill the Edit > Clear the scrollbar and viewport instead.  
<img width="400" alt="Screenshot 2024-11-22 at 11 46 51" src="https://github.com/user-attachments/assets/7db6ca62-241f-4ed5-9009-ff678454a22a">

Thanks.

**Comment by bew** [comment]
So what do you want exactly?
You've mentioned you want to bind cmd+K to clear viewport and scrollback, but just then shows that the keybind is correctly taken in account in menus.
What is the issue then, is it not working when you try the keybind?

**Reply by wez** [reply]
Maybe they want to use lower-case `k` for the keybind?

**Reply by bentinata** [reply]
@bew Sorry, but I think I've made myself clear? I want to reassign `Cmd+K` from just clear scrollback to clear scrollback and viewport. It is not correctly taken in the menus. The expected menu go towards another hotkey instead.

Would it be clearer if I remove the first screenshot and showing how defaults menu looks in macOS?

**Comment by bew** [comment, marked as answer]
Re-visiting old tabs I had still opened 🙄

Try `key = "k"` (lowercase), I think it'll do what you want (replace the keybind to not do simple clear but also clear scrollback).

I think the main issue you're shown in the screenshot is that MacOS menus are extremely confusing, as all characters are shown in uppercase even though it's actually the lowercase version that is bound.

**Reply by bentinata** [reply]
Thanks @bew. It works! I don't know why but I assume it does not differentiate uppercase and lowercase.

**Reply by bew** [reply]
MacOS does differentiate, it's just that it always shows the uppercase letter even for lowercase. And when the actual uppercase letter is bound to an action, the shift symbol is added in the menu next to the uppercase letter..

Very confusing..
